scheduler/reconcile.go

```diff
@@ -953,9 +954,30 @@ func (a *allocReconciler) handleDelayedLost(rescheduleLater []*delayedReschedule
 			// Set the evalID for the first alloc in this new batch
 			allocIDToFollowupEvalID[allocReschedInfo.allocID] = eval.ID
 		}
+		emitRescheduleInfo(allocReschedInfo.alloc, eval)
 	}
 
 	a.result.desiredFollowupEvals[tgName] = evals
 
 	return allocIDToFollowupEvalID
 }
+
+// emitRescheduleInfo emits metrics about the reschedule decision of an evaluation. If a followup evaluation is
+// provided, the waitUntil time is emitted.
+func emitRescheduleInfo(alloc *structs.Allocation, followupEval *structs.Evaluation) {
+	// Emit short-lived metrics data point. Note, these expire and stop emitting after about a minute.
+	baseMetric := []string{"client", "allocs", "reschedule"}
```

Comment:
This name would make these fall under [Client Allocation Metrics](https://www.nomadproject.io/docs/operations/metrics-reference#allocation-metrics) that we emit on the clients only. We have few metrics in the scheduler outside the RPC handlers (something we want to fix), so I don't have a good example of a scheduler-internal one. Maybe `[]string{"scheduler", "allocs", "rescheduled"}`?

Reply:
fixed the name in https://github.com/hashicorp/nomad/pull/11790